core/src/main/java/hudson/PluginManager.java

```diff
@@ -1436,14 +1435,14 @@ public HttpResponse doPluginsSearch(@QueryParameter String query, @QueryParamete
                     if (query == null || query.isBlank()) {
                         return true;
                     }
-                    return StringUtils.containsIgnoreCase(plugin.name, query) ||
-                        StringUtils.containsIgnoreCase(plugin.title, query) ||
-                        StringUtils.containsIgnoreCase(plugin.excerpt, query) ||
+                    return (plugin.name != null && plugin.name.toLowerCase().contains(query.toLowerCase())) ||
+                        (plugin.title != null && plugin.title.toLowerCase().contains(query.toLowerCase())) ||
+                        (plugin.excerpt != null && plugin.excerpt.toLowerCase().contains(query.toLowerCase())) ||
                         plugin.hasCategory(query) ||
                         plugin.getCategoriesStream()
                             .map(UpdateCenter::getCategoryDisplayName)
-                            .anyMatch(category -> StringUtils.containsIgnoreCase(category, query)) ||
-                        plugin.hasWarnings() && query.equalsIgnoreCase("warning:");
+                            .anyMatch(category -> category != null && category.toLowerCase().contains(query.toLowerCase())) ||
+                        (plugin.hasWarnings() && query.equalsIgnoreCase("warning:"));
```

Comment:
Seems like an unrelated change which could be removed to shrink the size of the diff and therefore its readability.

Reply:
already revert it.